inference-engine/tests/functional/plugin/cpu/single_layer_tests/roialign.cpp

```diff
@@ -0,0 +1,178 @@
+        selectedType = std::string("unknown_") + inPrc.name();
+        threshold = 0.001f;
+        const ngraph::ResultVector results{std::make_shared<ngraph::opset3::Result>(roialign)};
+        function = std::make_shared<ngraph::Function>(results, params, "ROIAlign");
+    }
+};
+
+TEST_P(ROIAlignLayerCPUTest, CompareWithRefs) {
+    SKIP_IF_CURRENT_TEST_IS_DISABLED()
+    Run();
+    CheckCPUImpl(executableNetwork, "ROIAlign");
+}
+
+namespace {
+
+/* CPU PARAMS */
+std::vector<CPUSpecificParams> filterCPUInfoForDevice() {
+    std::vector<CPUSpecificParams> resCPUParams;
+    if (with_cpu_x86_avx512f()) {
+        resCPUParams.push_back(CPUSpecificParams{{nchw, nc, x}, {nchw}, {}, {}});
+        resCPUParams.push_back(CPUSpecificParams{{nhwc, nc, x}, {nhwc}, {}, {}});
+        resCPUParams.push_back(CPUSpecificParams{{nChw16c, nc, x}, {nChw16c}, {}, {}});
+    } else if (with_cpu_x86_avx2() || with_cpu_x86_sse42()) {
+        resCPUParams.push_back(CPUSpecificParams{{nchw, nc, x}, {nchw}, {}, {}});
+        resCPUParams.push_back(CPUSpecificParams{{nhwc, nc, x}, {nhwc}, {}, {}});
+        resCPUParams.push_back(CPUSpecificParams{{nChw8c, nc, x}, {nChw8c}, {}, {}});
+    } else {
+        resCPUParams.push_back(CPUSpecificParams{{nchw, nc, x}, {nchw}, {}, {}});
+        resCPUParams.push_back(CPUSpecificParams{{nhwc, nc, x}, {nhwc}, {}, {}});
+    }
+    return resCPUParams;
+}
+
+const std::vector<InferenceEngine::Precision> netPrecisions = {
+        InferenceEngine::Precision::FP32,
+        InferenceEngine::Precision::BF16
+};
+
+const std::vector<int> spatialBinXVector = { 2 };
+
+const std::vector<int> spatialBinYVector = { 2 };
+
+const std::vector<float> spatialScaleVector = { 1.0f };
+
+const std::vector<int> poolingRatioVector = { 7 };
+
+const std::vector<std::string> modeVector = {
+        "avg",
+        "max"
+};
+
+const std::vector<std::vector<size_t>> inputShapeVector = {
+        SizeVector({ 1, 10, 20, 20 })
+};
+
+const std::vector<std::vector<float>> proposalsVector = {
+        { 1, 1, 19, 19 }
+};
+
+const std::vector<std::vector<size_t>> roisIdxVector = {
+        { 0 }
+};
+
+const auto roiAlignParams = ::testing::Combine(
+        ::testing::ValuesIn(spatialBinXVector),       // bin's column count
+        ::testing::ValuesIn(spatialBinYVector),       // bin's row count
+        ::testing::ValuesIn(spatialScaleVector),      // scale for given region considering actual input size
+        ::testing::ValuesIn(poolingRatioVector),      // pooling ratio for bin
+        ::testing::ValuesIn(proposalsVector),         // coordinate vector: left_top_x, left_top_y, right_bottom_x, right_bottom_y
+        ::testing::ValuesIn(roisIdxVector),           // batch id's vector
+        ::testing::ValuesIn(modeVector),              // pooling mode
+        ::testing::ValuesIn(inputShapeVector)         // feature map shape
+);
+
+INSTANTIATE_TEST_CASE_P(smoke_ROIAlignLayoutTest, ROIAlignLayerCPUTest,
```

Comment:
May I ask you add more test cases? With different channel dimension for example

Reply:
Done